src/main/java/duke/Duke.java

```diff
@@ -594,30 +599,29 @@ public void showList(TaskList tl) {
          *
          * @param tl The tasklist to be printed.
          */
-        public void showSearchList(TaskList tl) {
+        public String showSearchList(TaskList tl) {
             if (tl.isEmpty()) {
-                System.out.println("There are no matching tasks uwu. "
+                return ("There are no matching tasks uwu. "
                         + "(just like how whoever wrote this has no friends.)");
             } else {
-                System.out.println("Here are the matching tasks in your list uwu: ");
+                String toReturn = ("Here are the matching tasks in your list uwu: ");
                 int counter = 1;
                 for (Task t : tl.getList()) {
```

Comment:
I think under the Java coding standards the convention for a for loop is for (initialization; condition; update) (although I also use enhanced loops), you could consider changing it?